File: JAVA/Binary Tree Inorder Traversal.java
public class Solution {
    public ArrayList<Integer> inorderTraversal(TreeNode root) {
        ArrayList<Integer> ans = new ArrayList<Integer>();
        Stack<TreeNode> stack = new Stack<TreeNode>();
        TreeNode n = root;
        while (n!= null || !stack.empty()) {
            if (n!= null) {
                stack.push(n);
                n = n.left;
            }
            else {
                n = stack.pop();
                ans.add(n.val);
                n = n.right;
            }
        }
        return ans;
    }
}

File: Binary Tree Inorder Traversal.cpp
class Solution {
public:
    vector<int> inorderTraversal(TreeNode *root) {
        stack<TreeNode*> stk;
        vector<int> vec;
        TreeNode* node = root;
        while (not stk.empty() || node != NULL) {
            if (node != NULL) {
                stk.push(node);
                node = node->left;
            } else {
                node = stk.top();
                stk.pop();
                vec.push_back(node->val);
                node = node->right;
            }
        }
        return vec;
    }
};
